lib/components/diode.ts

```diff
@@ -1,6 +1,26 @@
 import { commonComponentProps, lrPolarPins } from "lib/common/layout"
-import type { z } from "zod"
+import { z } from "zod"
+
+const diodeConnectionKeys = z.enum([
+  "anode",
+  "cathode",
+  "pin1",
+  "pin2",
+  "pos",
+  "neg",
+])
+
+const connectionTarget = z
+  .string()
+  .or(z.array(z.string()).readonly())
+  .or(z.array(z.string()))
+
+const connectionsProp = z.record(diodeConnectionKeys, connectionTarget)
+
+export const diodeProps = commonComponentProps.extend({
+  connections: connectionsProp.optional(),
+})
 
-export const diodeProps = commonComponentProps.extend({})
 export const diodePins = lrPolarPins
+
 export type DiodeProps = z.input<typeof diodeProps>
```

Comment:
isn't this no longer how you're supposed to define props? Can we switch to the new style?

Reply:
The reason this is important is I want autocomplete to work properly for the diode connections object